src/test/java/org/sciborgs1155/robot/HopperTest.java

```diff
@@ -0,0 +1,45 @@
+package org.sciborgs1155.robot;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.sciborgs1155.lib.TestingUtil.*;
+import static org.sciborgs1155.robot.Hopper.HopperConstants.*;
+
+import edu.wpi.first.hal.HAL;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.sciborgs1155.robot.Hopper.Hopper;
+
+public class HopperTest {
+  Hopper hopper = new Hopper();
+
+  @BeforeEach
+  public void setup() {
+    assert HAL.initialize(5000, 0);
+  }
+
+  @Test
+  public void testMovementForward() {
```

Comment:
you can combine these first two tests into one testSetpoint test (or you can name it something else, doesn't really matter), where you run forwards and check the target speed, and then run back and check it again.